assets/queries/ansible/gcp/cloud_sql_instance_with_cross_db_ownership_chaining_on/query.rego

```diff
@@ -0,0 +1,38 @@
+package Cx
+
+CxPolicy [ result ] {
+  document := input.document[i]
+  tasks := getTasks(document)
+  task := tasks[t]
+  instance := task["google.cloud.gcp_sql_instance"]
+  contains(instance.database_version, "SQLSERVER")
+  settings := instance.settings
+  database_flags := settings.database_flags
+
+  check_database_flags_content(database_flags)
+
+  result := {
+                "documentId": 		document.id,
+                "searchKey": 	    sprintf("name=%s.{{google.cloud.gcp_sql_instance}}.settings.database_flags", [task.name]),
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue": "cloud_gcp_sql_instance.settings.database_flags are correct",
+                "keyActualValue": 	"cloud_gcp_sql_instance.settings.database_flags[0].name is 'cross db ownership chaining' and cloud_gcp_sql_instance.settings.database_flags[0].value is 'on'"
+              }
+}
+
+getTasks(document) = result {
+  result := document.playbooks[0].tasks
+} else = result {
+  object.get(document.playbooks[0],"tasks","undefined") == "undefined"
+  result := document.playbooks
+}
+
+check_database_flags_content(database_flags) {
+	database_flags[0].name == "cross db ownership chaining"
+	database_flags[0].value == "on"
```

Comment:
Here you are only checking for the first flag but what if the vulnerability is not in the first flag?
Try changing 'database_flags[0].name' and 'database_flags[0].value' to 'database_flags[x].name', 'database_flags[x].value'